spec/frameSpec.js

```diff
@@ -0,0 +1,207 @@
+/* eslint-env jasmine */
+const Frame = require('../src/frame');
```

Comment:
Your unit tests look pretty clean and well written - good. However, there do seem to be a lot of them! I think it's great that you're writing such detailed tests, but if I think of the cases that I think I want to test between game and frame, it's probably not much more than:

- a single roll produces the score for a single roll
- if I roll again, that roll gets assigned to the next available slot
- once my turn is complete, the next frame becomes active
- if I fail to knock down the pins, there is no bonus
- if I half strike, there is a single roll bonus
- if I strike, there are two roll bonuses
- the overall score calculates the sum of the frames
- if I need a bonus on the last frame, I get the correct number of relevant rolls

I may have missed a couple of cases, but not too much. So if I found myself writing a lot more tests than that, I might start thinking 'is my code properly encapsulated?'.